### PR TITLE
Add support for Infineon XENSIV PAS CO2 sensor

### PR DESCRIPTION
**Pull request by martgras**
# What does this implement/fix?

Add support for Infineon XENSIV PAS CO2 sensor

https://www.infineon.com/cms/en/product/sensor/co2-sensors/pasco2v01/

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2036

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
  - platform: pas_co2
    co2:
      name: "CO2"
      id: co2
    ambient_pressure_compensation: 1031
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/pas_co2/pas_co2.cpp [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+    ESP_LOGCONFIG(TAG, "  Ambient pressure compensation: %d hPa", this->ambient_pressure_);
+  }
+  ESP_LOGCONFIG(TAG, "  Calibration offset: %d ppm", this->calib_ref_);
+
+  LOG_SENSOR("  ", "CO2", this->co2_sensor_);
+}
+
+// Runs the sensor for 3 measurements on forced calibration mode
+// and then restores the prevoius settings
+void PasCo2Component::start_foced_calibration() {
```
Comment (jesserockz) [inline esphome/components/pas_co2/pas_co2.cpp:199]: typo?
```suggestion
void PasCo2Component::start_forced_calibration() {
```

File: esphome/components/pas_co2/pas_co2.h [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  void setup() override;
+  void update() override;
+  void dump_config() override;
+
+  void set_co2_sensor(sensor::Sensor *co2_sensor) { co2_sensor_ = co2_sensor; }
+  void set_ambient_pressure_source(sensor::Sensor *pressure) { ambient_pressure_source_ = pressure; }
+  void set_ambient_pressure_compensation(float pressure_in_hpa);
+  void set_calibration_offset(uint16_t offset) { this->calib_ref_ = offset; }
+  void set_abc_enable(bool enable) { this->enable_abc_ = enable; }
+  void start_foced_calibration();
```
Comment (jesserockz) [inline esphome/components/pas_co2/pas_co2.h:53]:
```suggestion
  void start_forced_calibration();
```

File: esphome/components/pas_co2/pas_co2.h [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  uint16_t calib_ref_{0};
+  bool enable_abc_{true};
+  bool initialized_{false};
+};
+
+template<typename... Ts> class PerformForcedCalibrationAction : public Action<Ts...> {
+ public:
+  PerformForcedCalibrationAction(PasCo2Component *pas_co2) : pas_co2_(pas_co2) {}
+
+  void play(Ts... x) override { this->pas_co2_->start_foced_calibration(); }
```
Comment (jesserockz) [inline esphome/components/pas_co2/pas_co2.h:100]:
```suggestion
  void play(Ts... x) override { this->pas_co2_->start_forced_calibration(); }
```

File: esphome/components/pas_co2/sensor.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+)
+
+CODEOWNERS = ["@martgras"]
+DEPENDENCIES = ["i2c"]
+
+pas_co2_ns = cg.esphome_ns.namespace("pas_co2")
+PasCo2Component = pas_co2_ns.class_(
+    "PasCo2Component", cg.PollingComponent, i2c.I2CDevice
+)
+PasCo2BackgroundCalibrationAction = pas_co2_ns.class_(
```
Comment (jesserockz) [inline esphome/components/pas_co2/sensor.py:22]:
```suggestion
PerformForcedCalibrationAction = pas_co2_ns.class_(
```

File: esphome/components/pas_co2/sensor.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+CALIBRATION_ACTION_SCHEMA = maybe_simple_id(
+    {
+        cv.Required(CONF_ID): cv.use_id(PasCo2Component),
+    }
+)
+
+
+@automation.register_action(
+    "pas_co2.perform_forced_calibration",
+    PasCo2BackgroundCalibrationAction,
```
Comment (jesserockz) [inline esphome/components/pas_co2/sensor.py:90]:
```suggestion
    PerformForcedCalibrationAction,
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Comment by martgras** [comment]
My test sensor is not responding anymore. I'll have to check if it's code or hw related

**Pull request closed without merging**